app/models/visit.rb

```diff
@@ -9,6 +9,14 @@ class Visit < ActiveRecord::Base
   has_one :rejection, dependent: :destroy
   has_one :cancellation, dependent: :destroy
 
+  has_many :banned_visitors, lambda {
```

Comment:
How about keeping `visitors` as the main association as all of the visitors need to be loaded always and changing the methods to something like this:

```ruby
def banned_visitors
  if visitors.loaded?
    visitors.select(&:banned?)
  else
    visitors.banned
  end
end
```

That way when we eagerload the visitors on a visit the methods will use the eager loaded data instead of triggering more DB calls